Python/Product/PythonTools/PythonTools/Project/PytestConfigureInfoBar.cs

```diff
@@ -0,0 +1,290 @@
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PythonTools.Infrastructure;
+using Microsoft.PythonTools.Interpreter;
+using Microsoft.PythonTools.Logging;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using OutputWindowRedirector = Microsoft.VisualStudioTools.Infrastructure.OutputWindowRedirector;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.PythonTools.Project {
+    /*
+     * Bugs: 
+     *      If an info bar is shown and then a user selects pytest from project settings->Tests->pytest, then the info bar will remain
+     *      If the user adds a config file after the project is loaded, the info bar will not be displayed
+     */
+    internal abstract class PyTestInfoBar : PythonInfoBar {
+        private PyTestInfoBarData InfoBarData { get; set; }
+
+        protected PyTestInfoBar(IServiceProvider site) : base(site) {
+
+        }
+
+        protected async Task CheckAsync(PyTestInfoBarData infoBarData) {
+            InfoBarData = infoBarData;
+            if (IsCreated || InfoBarData.IsGloballySuppressed || InfoBarData.InfoBarSuppressed) {
+                return;
+            }
+
+            /*
+            There are 4 possible cases. 
+            #1: Config file is     present, pytest is     installed, pytest is not enabled -> "Enable Pytest"
+            #2: Config file is     present, pytest is not installed, pytest is not enabled -> "Install and enable Pytest"
+            #3: Config file is     present, pytest is not installed, pytest is     enabled -> "Install Pytest"
+            #4: Config file is not present, pytest is not installed, pytest is     enabled -> "Install Pytest"
+             */
+            bool validConfigFile = File.Exists(infoBarData.PyTestConfigFilePath);
+
+            string infoBarMessage;
+            InfoBarHyperlink acceptActionItem;
+
+            if (validConfigFile &&
+                InfoBarData.TestFramework != TestFrameworkType.Pytest
+                && await IsPyTestInstalled()
+            ) {
+                //Case #1. "Enable Pytest"
+                infoBarMessage = Strings.PyTestInstalledConfigurationFileFound.FormatUI(InfoBarData.Caption, InfoBarData.Context.ToLower());
```

Comment:
InfoBarData.Context is initialized with a non-localized string that is used for telemetry purpose (which is why it is no localized). You can't insert that into a localized string. You'll have to add 2 more localizable strings to strings.resx for "workspace" and "project" and use those here (and everywhere else you used Context in FormatUI calls.